src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java

```diff
@@ -237,7 +264,52 @@ public void insertRecordToBuffer(SinkRecord kafkaSinkRecord) {
       if (copiedStreamingBuffer != null) {
         insertBufferedRecords(copiedStreamingBuffer);
       }
+    } else {
+      LOGGER.debug(
+          "Skip adding offset:{} to buffer for channel:{} because"
+              + " offsetPersistedInSnowflake:{}, processedOffset:{}",
+          kafkaSinkRecord.kafkaOffset(),
+          this.getChannelName(),
+          this.offsetPersistedInSnowflake.get(),
+          this.processedOffset.get());
+    }
+  }
+
+  /**
+   * If kafka offset was recently reset, we will skip adding any more records to buffer until we see
+   * a desired offset from kafka.
+   *
+   * <p>Desired Offset from Kafka = (offset persisted in snowflake + 1)
+   *
+   * <p>Check {link {@link TopicPartitionChannel#resetChannelMetadataAfterRecovery}} for reset logic
+   *
+   * @param kafkaSinkRecord Record to check for above condition only in case of failures
+   *     (isOffsetResetInKafka = true)
+   * @return true if this record can be skipped to add into buffer, false otherwise.
+   */
+  private boolean shouldIgnoreAddingRecordToBuffer(SinkRecord kafkaSinkRecord) {
+    // we should not buffer any record whose value is more than what we reset in kafka in case of
+    // failures
+    if (this.isOffsetResetInKafka.get()
```

Comment:
Reformat suggestion:
if (!this.isOffsetResetInKafka.get())
{
        return false; (or true?)
}

if kafkaSinkRecord.kafkaOffset() - offsetPersistedInSnowflake.get()) == 1L
{
        do something
}
else
{
        ignore
}

Reply:
Sure, can do this!